pymc3/distributions/multivariate.py

```diff
@@ -1781,119 +1797,87 @@ class KroneckerNormal(Continuous):
-            self._cov_type = "evd"
-            self._setup_evd(evds)
-
-    def _setup_evd(self, eigh_iterable):
-        self._isEVD = True
-        eigs_sep, Qs = zip(*eigh_iterable)  # Unzip
-        self.Qs = list(map(at.as_tensor_variable, Qs))
-        self.QTs = list(map(at.transpose, self.Qs))
-
-        self.eigs_sep = list(map(at.as_tensor_variable, eigs_sep))
-        self.eigs = kron_diag(*self.eigs_sep)  # Combine separate eigs
-        if self.is_noisy:
-            self.eigs += self.sigma ** 2
-        self.N = self.eigs.shape[0]
-
-    def _setup_random(self):
-        if not hasattr(self, "mv_params"):
-            self.mv_params = {"mu": self.mu}
-            if self._cov_type == "cov":
-                cov = kronecker(*self.covs)
-                if self.is_noisy:
-                    cov = cov + self.sigma ** 2 * at.identity_like(cov)
-                self.mv_params["cov"] = cov
-            elif self._cov_type == "chol":
-                if self.is_noisy:
-                    covs = []
-                    for eig, Q in zip(self.eigs_sep, self.Qs):
-                        cov_i = at.dot(Q, at.dot(at.diag(eig), Q.T))
-                        covs.append(cov_i)
-                    cov = kronecker(*covs)
-                    if self.is_noisy:
-                        cov = cov + self.sigma ** 2 * at.identity_like(cov)
-                    self.mv_params["chol"] = self.cholesky(cov)
-                else:
-                    self.mv_params["chol"] = kronecker(*self.chols)
-            elif self._cov_type == "evd":
+                eigh_iterable = eigh_map
                 covs = []
-                for eig, Q in zip(self.eigs_sep, self.Qs):
+                for eig, Q in zip(eigs_sep, Qs):
                     cov_i = at.dot(Q, at.dot(at.diag(eig), Q.T))
                     covs.append(cov_i)
                 cov = kronecker(*covs)
-                if self.is_noisy:
-                    cov = cov + self.sigma ** 2 * at.identity_like(cov)
-                self.mv_params["cov"] = cov
+                if is_noisy:
+                    cov = cov + sigma ** 2 * at.identity_like(cov)
+                chol = cholesky(cov)
+            else:
+                chol_diags = list(map(at.diag, chols))
+                sizes = at.as_tensor_variable([chol.shape[0] for chol in chols])
+                N = at.prod(sizes)
+                chol = kronecker(*chols)
+            # Check if this is the correct operation
+            cov = quaddist_matrix(cov, chol)
+        else:
+            eigh_iterable = evds
+            covs = []
+            for eig, Q in zip(eigs_sep, Qs):
+                cov_i = at.dot(Q, at.dot(at.diag(eig), Q.T))
+                covs.append(cov_i)
+            cov = kronecker(*covs)
+            if is_noisy:
+                cov = cov + sigma ** 2 * at.identity_like(cov)
 
-    def random(self, point=None, size=None):
+        mu = at.as_tensor_variable(mu)
+
+        chols = at.as_tensor_variable(chols)
```

Comment:
@ricardoV94 The `chols` parameter here is a list of either `Cholesky` Ops or a pre-defined list of arrays. How do we turn this into the `TensorConstant` variant ?

Reply:
I don't think we can make them Constants, it seems like everything can be symbolic here. Can't we stack the `chols`?

Reply:
The list gets converted into a `np.array` with `dtype` as `object` and `TensorVariable` doesn't support that.